packages/clerk-js/src/ui/utils/colors/index.ts

```diff
@@ -0,0 +1,87 @@
+import type { HslaColor, HslaColorString } from '@clerk/types';
+
+import { hasModernColorSupport } from './cache';
+import { legacyColors } from './legacy';
+import { modernColors } from './modern';
+
+/**
+ * Unified colors API that automatically chooses between modern and legacy implementations
+ * based on browser support. Maintains the same interface as the legacy colors object.
+ */
+export const colors = {
```

Comment:
if we define this as `export const colors = hasModernColorSupport() ? {/* implementation with modernColors */} : {/* implementation with legacyColors */};` we can skip the cost of checking the cache for each operation.